### PR TITLE
dev/echo/support rescheduling

### PR DESCRIPTION
**Pull request by XinyiSong**

### CODE REVIEW

File: pkg/framework/api/common_state.go [hunk 1]
```diff
@@ -51,6 +53,12 @@ const (
 	PodPropertyKey = "PodProperty"
 
 	AssignedNumasKey = "AssignedNumas"
+
+	PodSchedulingStageStateKey = "PodSchedulingStageState"
+
+	NotScheduledPodKeysByTemplateKey = "NotScheduledPodKeysByTemplate"
+
+	PostNodeCircleSchedulingFuncKey = "PostNodeCircleSchedulingFunc"
```
Comment (binacs) [inline pkg/framework/api/common_state.go:61]: nit: this is a temporary solution (for metrics) and should not be migrated to the open source repo

File: pkg/framework/api/movement_test.go [hunk 2]
```diff
@@ -0,0 +1,840 @@
...
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+	"testing"
+
+	schedulingv1a1 "github.com/kubewharf/godel-scheduler-api/pkg/apis/scheduling/v1alpha1"
+	"github.com/kubewharf/godel-scheduler/pkg/util/generationstore"
+	podutil "github.com/kubewharf/godel-scheduler/pkg/util/pod"
```
Comment (binacs) [inline pkg/framework/api/movement_test.go:11]: nit: format this (separate from k8s pkg)

**Review by binacs — COMMENTED** [review]
just some nits, and it seems that rebase is needed

File: pkg/framework/api/common_state.go [hunk 3]
```diff
@@ -297,3 +305,44 @@ func GetAssignedNumas(state *CycleState) ([]int, error) {
...
+
+	if data, err := state.Read(PostNodeCircleSchedulingFuncKey); err == nil {
+		if s, ok := data.(*stateData); ok {
+			return s.data.(PostNodeCircleSchedulingFunc), nil
+		}
+	}
+	return nil, fmt.Errorf("post nodeCircle scheduling func not found")
+}
+
+func RunPostNodeCircleSchedulingFunc(unitState *CycleState, nodeCircle NodeCircle, pod *v1.Pod, suggesstHost string, err error) {
```
[the remark this answers is not in the document]
Comment (XinyiSong) [reply pkg/framework/api/common_state.go]: fixed

File: pkg/framework/api/common_state.go (hunk 1 above)
Comment (XinyiSong) [reply pkg/framework/api/common_state.go]: fixed

**Review by rookie0080 — APPROVED** [review]

**Review by binacs — APPROVED** [review]
lgtm

**Pull request merged**